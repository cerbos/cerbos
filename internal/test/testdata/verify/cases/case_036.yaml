description: Time-based conditions without specifying `now`
