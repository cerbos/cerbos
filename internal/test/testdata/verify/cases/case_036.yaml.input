-- testdata/principals.yaml --
---
principals:
  bev:
    id: bev
    policyVersion: '20210210'
    roles:
      - employee
      - manager
    attr:
      department: marketing
      geography: GB
      managed_geographies: GB
      ip_address: 10.20.1.2
      team: design

-- testdata/resources.yaml --
---
resources:
  stale_pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design
      modifiedAt: "2022-08-01T15:00:00Z"
      status: PENDING_APPROVAL

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- suite_test.yaml --
---
name: TestSuite
description: Tests for verifying something
tests:
  - name: Without now
    input:
      principals:
        - bev
      resources:
        - stale_pending_leave_request
      actions:
        - remind
      auxData: myJWT
    expected:
      - principal: bev
        resource: stale_pending_leave_request
        actions:
          remind: EFFECT_ALLOW
