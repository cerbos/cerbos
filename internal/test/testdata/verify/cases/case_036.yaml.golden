{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_ERRORED",
        "testsCount": 1,
        "resultCounts": [
          {
            "result": "RESULT_ERRORED",
            "count": 1
          }
        ]
      },
      "testCases": [
        {
          "name": "Without now",
          "principals": [
            {
              "name": "bev",
              "resources": [
                {
                  "name": "stale_pending_leave_request",
                  "actions": [
                    {
                      "name": "remind",
                      "details": {
                        "result": "RESULT_ERRORED",
                        "error": "a policy used a time-based condition, but `now` was not provided in the test options"
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Tests for verifying something"
    }
  ],
  "summary": {
    "overallResult": "RESULT_ERRORED",
    "testsCount": 1,
    "resultCounts": [
      {
        "result": "RESULT_ERRORED",
        "count": 1
      }
    ]
  }
}
